pkg/da/data_availability_header_test.go

```diff
@@ -239,6 +252,70 @@ func TestSquareSize(t *testing.T) {
 	}
 }
 
+func TestConstructEDS_Versions(t *testing.T) {
+	for version := uint64(0); version <= appconsts.Version*uint64(2); version++ {
```

Comment:
[nit] we probably only need to go 1 above the current app version. Otherwise, half the iterations in this loop will be error cases.

```suggestion
	for version := uint64(0); version <= appconsts.Version+uint64(1); version++ {
```